1 - Faça um Programa que peça dois números e imprima o maior deles.

In [31]:
n1 = input( "Digite um valor: " )
n2 = input( "Digite outro valor:" )
if ( n1>n2 ):
    print( "MAIOR VALOR: ", n1 )
elif ( n2>n1 ):
    print( "MAIOR VALOR: ", n2 )
else:
    print( "VALORES IGUAIS!" )

Digite um valor: 15
Digite outro valor:15
Os valores são iguais!


2 - Faça um Programa que verifique se uma letra digitada é vogal ou consoante.

In [ ]:
lt = input( "Digite uma letra a seguir: " )
if( lt.lower(  ) in ['a', 'e', 'i', 'o', 'u'] ):
    print( "A letra é vogal" )
elif( lt.lower(  ) in ['á', 'à', 'ã', 'â', 'ä', 'é', 'è', 'ê', 'ë', 'í', 'ì', 'î', 'ï', 'ó', 'ò', 'ô', 'õ', 'ö', 'ú', 'ù', 'û', 'ü'] ):
    print(  "É vogal com acento" )
else:
    print( "A letra é consoante" )

3 - Faça um programa para a leitura de duas notas parciais de um aluno. O programa deve calcular a média alcançada por aluno e apresentar:
- A mensagem "Aprovado", se a média alcançada for maior ou igual a sete;
- A mensagem "Reprovado", se a média for menor do que sete;
- A mensagem "Aprovado com Distinção", se a média for igual a dez.

In [ ]:
n1 = float( input( "Digite sua primeira nota: " ) )
n2 = float( input( "Digite sua segunda nota: " ) )
media = ( n1+n2 )/2
if media<7:
    print( "Reprovado: ", media )
else:
    if media==10:
        print( "Aprovado com Distinção: ", media )
    else:
        print( "Aprovado: ", media )

4 - Faça um Programa que leia três números e mostre-os em ordem decrescente.

In [ ]:
n1 = float( input( "Digite um número: " ) )
n2 = float( input( "Digite outro número: " ) )
n3 = float( input( "Digite um terceiro número: " ) )
lista = [n1, n2, n3]
s_list = sorted( lista, reverse=True )
print( "Os números na sequência ficam: ", s_list )

5 - As Organizações Tabajara resolveram dar um aumento de salário aos seus colaboradores e lhe contrataram para desenvolver o programa que calculará os reajustes.
Faça um programa que recebe o salário de um colaborador e o reajuste segundo o seguinte critério, baseado no salário atual:
- salários até R$ 280,00 (incluindo) : aumento de 20%
- salários entre R$ 280,00 e R$ 700,00 : aumento de 15%
- salários entre R$ 700,00 e R$ 1500,00 : aumento de 10%
- salários de R$ 1500,00 em diante : aumento de 5% Após o aumento ser realizado, informe na tela:
- o salário antes do reajuste;
- o percentual de aumento aplicado;
- o valor do aumento;
- o novo salário, após o aumento.

In [ ]:
salario = float( input( "Informe o salário do funcionário: " ) )
percentual = 0.0
if( salario <= 280 ):
    salario_reajustado = salario + ( salario*20/100 )
    percentual = 20
else:
    if( salario <= 700 ):
        salario_reajustado = salario + ( salario*15/100 )
        percentual = 15
    else:
        if( salario <= 1500 ):
            salario_reajustado = salario + ( salario*10/100 )
            percentual = 10
        else:
            salario_reajustado = salario + ( salario*5/100 )
            percentual = 5
print( f"O salário antes do reajuste era: {salario:.2f}\nO percentual aplicado foi de: {percentual:.2f}%\nO aumento foi de {salario_reajustado-salario:.2f}\nO novo salário é: {salario_reajustado:.2f}" )

6 - Faça um Programa que leia um número e exiba o dia correspondente da semana. (1-Domingo, 2- Segunda, etc.), se digitar outro valor deve aparecer valor inválido.

In [ ]:
dici = {'1':'Domingo', '2':'Segunda', '3':'Terça', '4':'Quarta', '5':'Quinta', '6':'Sexta', '7':'Sábado'}
day = int( input( "Digite um número da semana: " ) )
if ( day>0 and day<=7 ):
    print( dici[str( day )] )
else:
    print( "Número inválido" )

7 - Faça um programa que lê as duas notas parciais obtidas por um aluno numa disciplina ao longo de um semestre, e calcule a sua média. A atribuição de conceitos obedece à tabela abaixo:
  Média de Aproveitamento  Conceito
  Entre 9.0 e 10.0        A
  Entre 7.5 e 9.0         B
  Entre 6.0 e 7.5         C
  Entre 4.0 e 6.0         D
  Entre 4.0 e zero        E
O algoritmo deve mostrar na tela as notas, a média, o conceito correspondente e a mensagem “APROVADO” se o conceito for A, B ou C ou “REPROVADO” se o conceito for D ou E.

In [ ]:
n1 = float( input( "Digite a primeira nota: " ) )
n2 = float( input( "Digite a segunda nota: " ) )
media = ( n1+n2 )/2
nota = ''
status = ''
if ( media >= 9 and media <= 10 ):
    nota = 'A'
else:
    if ( media >= 7.5 and media < 9 ):
        nota = 'B'
    else:
        if ( media >= 6 ):
            nota  = 'C'
        else:
            if ( media >= 4 ):
                nota = 'D'
            else:
                if ( media <4 and media >= 0 ):
                    nota = 'E'
if ( nota == 'A' or nota == 'B' or nota == 'C' ):
    status = 'APROVADO'
else:
    status = 'REPROVADO'
print ( f"{status} com conceito {nota}" )

8 - Faça um Programa que peça os 3 lados de um triângulo. O programa deverá informar se os valores podem ser um triângulo. Indique, caso os lados formem um triângulo, se o mesmo é: equilátero, isósceles ou escaleno.
Dicas:
- Três lados formam um triângulo quando a soma de quaisquer dois lados for maior que o terceiro;
- Triângulo Equilátero: três lados iguais;
- Triângulo Isósceles: quaisquer dois lados iguais;
- Triângulo Escaleno: três lados diferentes;

In [ ]:
a = float( input( "Digite um lado do triângulo: " ) )
b = float( input( "Digite outro lado do triângulo: " ) )
c = float( input( "Digite o terceiro lado do triângulo: " ) )
if ( b-c < a and a < b+c ):
    if ( a-c < b and b <a+c ):
        if ( a-b < c < a+b ):
            if ( a == b and b == c ):
                print( "É um triângulo Equilátero" )
            elif ( a == b or a == c or b == c ):
                print( "É um triângulo Isóscceles" )
            else:
                print( "É um triângulo Escaleno" )
        else:
            print( "Não é um triângulo" )
else:
    print( "Não é um triângulo" )

9 - Faça um programa que calcule as raízes de uma equação do segundo grau, na forma ax2 + bx + c. O programa deverá pedir os valores de a, b e c e fazer as consistências, informando ao usuário nas seguintes situações:
- Se o usuário informar o valor de A igual a zero, a equação não é do segundo grau e o programa não deve fazer pedir os demais valores, sendo encerrado;
- Se o delta calculado for negativo, a equação não possui raizes reais. Informe ao usuário e encerre o programa;
- Se o delta calculado for igual a zero a equação possui apenas uma raiz real; informe-a ao usuário;
- Se o delta for positivo, a equação possui duas raiz reais; informe-as ao usuário;

In [ ]:
import sys
delta = 0
a = float( input( "Digite o valor de A: " ) )
if( a == 0 ):
    print( "A equação não é de segundo grau" )
    sys.exit( "A equação não é de segundo grau" )
else:
    b = float( input( "Digite o valor de B: " ) )
    c = float( input( "Digite o valor de C: " ) )
delta = b**2 - 4*a*c
x1 = ( -b + delta**( 1/2 ) ) / ( 2*a )
x2 = ( -b - delta**( 1/2 ) ) / ( 2*a )
if( delta < 0 ):
    print( "A equação não possui raízes reais" )
    sys.exit( "A equação não possui raízes reais" )
elif( delta == 0 ):
    print( f"A equação possui apenas um raiz real: {x1:.2f}" )
else:
    print( f"A equação possui duas raízes reais, x1 = {x1} e x2 = {x2}" )

10 - Faça um Programa para um caixa eletrônico. O programa deverá perguntar ao usuário o valor do saque e depois informar quantas notas de cada valor serão fornecidas. As notas disponíveis serão as de 1, 5, 10, 50 e 100 reais. O valor mínimo é de 10 reais e o máximo de 600 reais. O programa não deve se preocupar com a quantidade de notas existentes na máquina.
- Exemplo 1: Para sacar a quantia de 256 reais, o programa fornece duas notas de 100, uma nota de 50, uma nota de 5 e uma nota de 1;
- Exemplo 2: Para sacar a quantia de 399 reais, o programa fornece três notas de 100, uma nota de 50, quatro notas de 10, uma nota de 5 e quatro notas de 1.

In [ ]:
saque = float( input( "Digite um valor de saque, mínimo R$ 10.00, máximo R$ 600.00: " ) )
restante = saque
n1 = 0
n5 = 0
n10 = 0
n50 = 0
n100 = 0
if ( saque < 10 or saque > 600 ):
    sys.exit( "O valor não está dentro do máximo ou mínimo estipulado" )
else:
    while( restante>=100 ):
        n100+=1
        restante-=100
    while( restante>=50 ):
        n50+=1
        restante-=50
    while( restante>=10 ):
        n10+=1
        restante-=10
    while( restante>=5 ):
        n5+=1
        restante-=5
    while( restante>=1 ):
        n1+=1
        restante-=1
print( f"Serão fornecidas para o sauqe total de: {saque:.2f}\n{n100} nota( s ) de R$100,00\n{n50} nota( s ) de R$50,00\n{n10} nota( s ) de R$10,00\n{n50} nota( s ) de R$5,00\n{n1} nota( s ) de R$1,00" )

11 - Faça um programa que faça 5 perguntas para uma pessoa sobre um crime. As perguntas são:
"Telefonou para a vítima?"
"Esteve no local do crime?"
"Mora perto da vítima?"
"Devia para a vítima?"
"Já trabalhou com a vítima?" 
O programa deve no final emitir uma classificação sobre a participação da pessoa no crime. Se a pessoa responder positivamente a 2 questões ela deve ser classificada como "Suspeita", entre 3 e 4 como "Cúmplice" e 5 como "Assassino". Caso contrário, ele será classificado como "Inocente".

In [14]:
import sys
perguntas = {
    '1': 'Telefonou para a vítima? ',
    '2': 'Esteve no local do crime? ',
    '3': 'Mora perto da vítima? ',
    '4': 'Devia para a vítima? ',
    '5': 'Já trabalhou com a vítima? '
}
r = ''
respostas = []
qtde = 0
print( "Responda 's' para SIM e 'n' para NÃO para as próximas perguntas\n" )
for x,y in perguntas.items(  ):
    r = input( y )
    if( r == 's' or r == 'n' ):
        respostas += [r]
    else:
        sys.exit( "Só é aceito s ou n" )
for x in respostas:
    if( x=='s' ):
        qtde += 1
if( qtde == 2 ):
    print( "Suspeita" )
elif( qtde == 3 or qtde == 4 ):
    print( "Cúmplice" )
elif( qtde == 5 ):
    print( "Assassino" )
else:
    print( "Inocente" )

Responda 's' para SIM e 'n' para NÃO para as próximas perguntas

Telefonou para a vítima? n
Esteve no local do crime? s
Mora perto da vítima? n
Devia para a vítima? s
Já trabalhou com a vítima? s
Cúmplice


12 - Uma fruteira está vendendo frutas com a seguinte tabela de preços:

                      Até 5 Kg           Acima de 5 Kg
Morango         R$ 2,50 por Kg          R$ 2,20 por Kg
Maçã            R$ 1,80 por Kg          R$ 1,50 por Kg

Se o cliente comprar mais de 8 Kg em frutas ou o valor total da compra ultrapassar R$ 25,00, receberá ainda um desconto de 10% sobre este total. Escreva um algoritmo para ler a quantidade (em Kg) de morangos e a quantidade (em Kg) de maças adquiridas e escreva o valor a ser pago pelo cliente.

In [ ]:
morango = float( input( "Digite quantos Kg de MORANGO deseja: " ) )
maca = float( input( "Digite quantos Kg de MAÇÃ deseja: " ) )
v_morango = 0.0
v_maca = 0.0
if( morango <= 5 ):
    v_morango = morango*2.5
else:
    v_morango = morango*2.2
    if( morango >= 8 or v_morango >= 25 ):
        v_morango = v_morango - ( 10/100 * v_morango )
if( maca <= 5 ):
    v_maca = maca*1.8
else:
    v_maca = maca*1.5
    if( maca >= 8 or v_maca >= 25 ):
        v_maca = v_maca - ( 10/100 * v_maca )
print( f"O valor do Morango é: R$ {v_morango:.2f}\nO valor da Maçã é: R$ {v_maca:.2f}" )

13 - Faça um programa que peça uma nota, entre zero e dez. Mostre uma mensagem caso o valor seja inválido e continue pedindo até que o usuário informe um valor válido.

In [7]:
nota = float( input( "Digite uma nota entre 0 e 10: " ) )
while( nota < 0 or nota > 10 ):
    nota = float( input( "O valor é inválido, digite um novo valor: " ) )
print( "Valor válido", nota )

Digite uma nota entre 0 e 10: 5
Valor válido 5.0


In [58]:
nome = input( "Digite um nome com mais de 3 caracteres: " )
idade = int( input( "Digite uma idade entre 0 e 150 anos: " ) )
salario = float( input( "DIgite um salário maior que R$ 0,00: " ) )
sexo = input( "Digite 'f' para sexo FEMININO e para 'm' para MASCULINO: " )
e_civil = input( "Digite 's' para 'SOLTEIRO( A )', 'c' para 'CASADO( A )', 'v' para 'VIÚVO( A )' ou 'd' para 'DIVORCIADO( A )'" )
dici_civil = {
    's': 'SOLTEIRO( A )',
    'c': 'CASADO( A )',
    'v': 'VIÚVO( A )',
    'd': 'DIVORCIADO( A )'
}
if( len( nome ) > 3 ):
    print( "O nome possui mais que 3 caracteres" )
else:
    print( "O nome NÃO possui mais que 3 caracteres" )
if( idade >= 0 and idade <= 150 ):
    print( "A idade está entre 0 e 150 anos" )
else:
    print( "A idade NÃO está entre 0 e 150 anos" )
if( salario > 0 ):
    print( "O salário é maior que R$ 0.00" )
else:
    print( "O salário NÃO é maior que R$ 0.00" )
if( sexo == 'f' ):
    print( "Sexo feminino" )
elif( sexo == 'm' ):
    print( "Sexo masculino" )
else:
    print( "Sexo não tem formato válido" )
if( e_civil in dici_civil.keys(  ) ):
    print( "O Estado Civil selecionado é: ", dici_civil[e_civil] )
else:
    print( "O Estado Civil não correponde a um valor válido" )

Digite um nome com mais de 3 caracteres: 3
Digite uma idade entre 0 e 150 anos: 165
DIgite um salário maior que R$ 0,00: -4
Digite 'f' para sexo FEMININO e para 'm' para MASCULINO: a
Digite 's' para 'SOLTEIRO(A)', 'c' para 'CASADO(A)', 'v' para 'VIÚVO(A)' ou 'd' para 'DIVORCIADO(A)'w
O nome NÃO possui mais que 3 caracteres
A idade NÃO está entre 0 e 150 anos
O salário NÃO é maior que R$ 0.00
Sexo não tem formato válido
O Estado Civil não correponde a um valor válido


14 - Faça um programa que leia 5 números e informe o maior número.

In [62]:
lista = []
while(len(lista) < 5):
    n = float(input("Digite um número: "))
    lista += [n]
lista = sorted(lista, key=float)
print("O maior número é ", lista[4])

Digite um número: 8.2
Digite um número: 9.7
Digite um número: 4.56
Digite um número: 9
Digite um número: 2
O maior número é  9.7


15 - Faça um programa que leia 5 números e informe a soma e a média dos números.

In [63]:
lista = []
soma = 0.0
media = 0.0
while(len(lista) < 5):
    n = float(input("Digite um número: "))
    soma += n
    lista += [n]
media = soma / 5
print(f"A soma é = {soma}\nA média é = {media}")

Digite um número: 4
Digite um número: 8
Digite um número: 9.5
Digite um número: 4.2
Digite um número: 6.8
A soma é = 32.5
A média é = 6.5


16 - Faça um programa que imprima na tela apenas os números ímpares entre 1 e 50.

In [16]:
lista = [x for x in range(1,51) if x%2!=0]
print(lista)

[1, 3, 5, 7, 9, 11, 13, 15, 17, 19, 21, 23, 25, 27, 29, 31, 33, 35, 37, 39, 41, 43, 45, 47, 49]


17 - Desenvolva um gerador de tabuada, capaz de gerar a tabuada de qualquer número inteiro entre 1 a 10. O usuário deve informar de qual numero ele deseja ver a tabuada. A saída deve ser conforme o exemplo abaixo:
Tabuada de 5:
5 X 1 = 5
5 X 2 = 10
...
5 X 10 = 50

In [73]:
n = int(input("Digite um número para visualizar sua tabuada"))
lista = [x*n for x in range(1,11)]
for x in range(len(lista)):
    print(f"{n} X {x+1} = {lista[x]}")

Digite um número para visualizar sua tabuada4
4 X 1 = 4
4 X 2 = 8
4 X 3 = 12
4 X 4 = 16
4 X 5 = 20
4 X 6 = 24
4 X 7 = 28
4 X 8 = 32
4 X 9 = 36
4 X 10 = 40


18 - A série de Fibonacci é formada pela seqüência 1,1,2,3,5,8,13,21,34,55,... Faça um programa capaz de gerar a série até o n−ésimo termo.

In [99]:
n = int( input( "Quantos valores da série Fibonacci deseja mostrar?: " ) )

ultimo=1
penultimo=1
lista = []

if ( n==1 ):
    lista = [1]
elif ( n==2 ):
    lista = [1, 1]
else:
    lista = [1, 1]
    count=3
    while count <= n:
        termo = ultimo + penultimo
        penultimo = ultimo
        ultimo = termo
        lista += [termo]
        count += 1
    print( "O termo selecionado é: ", termo )
print( "A série fica: ", lista )

Quantos valores da série Fibonacci deseja mostrar?: 15
O termo selecionado é:  610
A série fica:  [1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233, 377, 610]


19 - Faça um programa que calcule o fatorial de um número inteiro fornecido pelo usuário. Ex.: 5!=5.4.3.2.1=120

In [109]:
n = int( input( "Digite um número para calcular seu fatorial: " ) )
lista = []
fatorial = n

lista = [x for x in range( 1,n )]
for x in range( len( lista ) ):
    fatorial = fatorial*lista[x]
print( f"O valor fatorial de {n}! = {fatorial}" )

Digite um número para calcular seu fatorial: 4
O valor fatorial de 4! = 24


20 - O Sr. Manoel Joaquim possui uma grande loja de artigos de R$ 1,99, com cerca de 10 caixas. Para agilizar o cálculo de quanto cada cliente deve pagar ele desenvolveu um tabela que contém o número de itens que o cliente comprou e ao lado o valor da conta. Desta forma a atendente do caixa precisa apenas contar quantos itens o cliente está levando e olhar na tabela de preços. Você foi contratado para desenvolver o programa que monta esta tabela de preços, que conterá os preços de 1 até 50 produtos, conforme o exemplo abaixo:

Lojas Quase Dois - Tabela de preços
1 - R$ 1.99
2 - R$ 3.98
...
50 - R$ 99.50

In [10]:
# itens = int(input("Digite quantos produtos o cliente deseja comprar: "))
valor = [x*1.99 for x in range(1,51)]
print("Lojas Quase Dois - Tabela de preços:")
for x in range(len(valor)):
    print(f"{x+1} - R$ {valor[x]:.2f}")
# print(valor, valor[itens-1])

Lojas Quase Dois - Tabela de preços:
1 - R$ 1.99
2 - R$ 3.98
3 - R$ 5.97
4 - R$ 7.96
5 - R$ 9.95
6 - R$ 11.94
7 - R$ 13.93
8 - R$ 15.92
9 - R$ 17.91
10 - R$ 19.90
11 - R$ 21.89
12 - R$ 23.88
13 - R$ 25.87
14 - R$ 27.86
15 - R$ 29.85
16 - R$ 31.84
17 - R$ 33.83
18 - R$ 35.82
19 - R$ 37.81
20 - R$ 39.80
21 - R$ 41.79
22 - R$ 43.78
23 - R$ 45.77
24 - R$ 47.76
25 - R$ 49.75
26 - R$ 51.74
27 - R$ 53.73
28 - R$ 55.72
29 - R$ 57.71
30 - R$ 59.70
31 - R$ 61.69
32 - R$ 63.68
33 - R$ 65.67
34 - R$ 67.66
35 - R$ 69.65
36 - R$ 71.64
37 - R$ 73.63
38 - R$ 75.62
39 - R$ 77.61
40 - R$ 79.60
41 - R$ 81.59
42 - R$ 83.58
43 - R$ 85.57
44 - R$ 87.56
45 - R$ 89.55
46 - R$ 91.54
47 - R$ 93.53
48 - R$ 95.52
49 - R$ 97.51
50 - R$ 99.50


21 - O Sr. Manoel Joaquim acaba de adquirir uma panificadora e pretende implantar a metodologia da tabelinha, que já é um sucesso na sua loja de 1,99. Você foi contratado para desenvolver o programa que monta a tabela de preços de pães, de 1 até 50 pães, a partir do preço do pão informado pelo usuário, conforme o exemplo abaixo:

Preço do pão: R$ 0.18
Panificadora Pão de Ontem - Tabela de preços
1 - R$ 0.18
2 - R$ 0.36
...
50 - R$ 9.00

In [1]:
valor = float(input("Digite o valor do pão: "))
tabela = [x*valor for x in range(1,51)]
print("Panificadora Pão de Ontem - Tabela de preços:")
for x in range(len(tabela)):
    print(f"{x+1} - R$ {tabela[x]:.2f}")

Panificadora Pão de Ontem - Tabela de preços:
1 - R$ 0.20
2 - R$ 0.40
3 - R$ 0.60
4 - R$ 0.80
5 - R$ 1.00
6 - R$ 1.20
7 - R$ 1.40
8 - R$ 1.60
9 - R$ 1.80
10 - R$ 2.00
11 - R$ 2.20
12 - R$ 2.40
13 - R$ 2.60
14 - R$ 2.80
15 - R$ 3.00
16 - R$ 3.20
17 - R$ 3.40
18 - R$ 3.60
19 - R$ 3.80
20 - R$ 4.00
21 - R$ 4.20
22 - R$ 4.40
23 - R$ 4.60
24 - R$ 4.80
25 - R$ 5.00
26 - R$ 5.20
27 - R$ 5.40
28 - R$ 5.60
29 - R$ 5.80
30 - R$ 6.00
31 - R$ 6.20
32 - R$ 6.40
33 - R$ 6.60
34 - R$ 6.80
35 - R$ 7.00
36 - R$ 7.20
37 - R$ 7.40
38 - R$ 7.60
39 - R$ 7.80
40 - R$ 8.00
41 - R$ 8.20
42 - R$ 8.40
43 - R$ 8.60
44 - R$ 8.80
45 - R$ 9.00
46 - R$ 9.20
47 - R$ 9.40
48 - R$ 9.60
49 - R$ 9.80
50 - R$ 10.00


22 - O Sr. Manoel Joaquim expandiu seus negócios para além dos negócios de 1,99 e agora possui uma loja de conveniências. Faça um programa que implemente uma caixa registradora rudimentar. O programa deverá receber um número desconhecido de valores referentes aos preços das mercadorias. Um valor zero deve ser informado pelo operador para indicar o final da compra. O programa deve então mostrar o total da compra e perguntar o valor em dinheiro que o cliente forneceu, para então calcular e mostrar o valor do troco. Após esta operação, o programa deverá voltar ao ponto inicial, para registrar a próxima compra. A saída deve ser conforme o exemplo abaixo:

Lojas Tabajara 
Produto 1: R$ 2.20
Produto 2: R$ 5.80
Produto 3: R$ 0
Total: R$ 9.00
Dinheiro: R$ 20.00
Troco: R$ 11.00
...

In [40]:
p_produtos = []
valor = -1
soma = 0
while(valor != 0):
    valor = float(input("Digite o valor do produto: R$ "))
    p_produtos += [valor]
    soma = soma + valor
print("Lojas Tabajara")
for x in range(len(p_produtos)):
    print(f"Produto {x+1}: R$ {p_produtos[x]}")
print(f"Total: R$ {soma:.2f}")
dinheiro = round(float(input(f"Dinheiro: R$ ")),2)
troco = dinheiro - soma
print(f"Troco: R$ {troco:.2f}")

Digite o valor do produto: R$ 5.2
Digite o valor do produto: R$ 3.6
Digite o valor do produto: R$ 0
Lojas Tabajara
Produto 1: R$ 5.2
Produto 2: R$ 3.6
Produto 3: R$ 0.0
Total: R$ 8.80
Dinheiro: R$ 20
Troco: R$ 11.20


23 - Foi feita uma estatística em cinco cidades brasileiras para coletar dados sobre acidentes de trânsito. Foram obtidos os seguintes dados:
- Código da cidade;
- Número de veículos de passeio (em 1999);
- Número de acidentes de trânsito com vítimas (em 1999). Deseja-se saber:
- Qual o maior e menor índice de acidentes de transito e a que cidade pertence;
- Qual a média de veículos nas cinco cidades juntas;
- Qual a média de acidentes de trânsito nas cidades com menos de 2.000 veículos de passeio.

In [135]:
n_cidades = 3

count = 0
count_2000 = 0
dados = {}
lista = []
sum_v = 0
sum_a = 0
media_v = 0
media_a = 0

def sortAcidente( val ): 
    return val[2]

while( count < n_cidades ):
    uf = input( "Digite o código da cidade: " )
    n_veiculo = int( input( "Digite o número de veículos de passeio ( em 1999 ): " ) )
    n_acidente = int( input( "Digite o número de acidentes de trânsito com vítimas ( em 1999 ): " ) )
    lista += [( uf, n_veiculo, n_acidente )]
    sum_v += + n_veiculo
    if( n_veiculo < 2000 ):
        sum_a += n_acidente
        count_2000 += 1
    count+=1
media_v = sum_v / n_cidades
media_a = sum_a / count_2000
print( lista )
lista.sort( key = sortAcidente, reverse = True )
print( f"O maior índice de acidentes de trânsito é de:\n -> {lista[0][2]} veículos em {lista[0][0]}" )
lista.sort( key = sortAcidente )
print( f"O menor índice de acidentes de trânsito é de:\n -> {lista[0][2]} veículos em {lista[0][0]}" )
print( f"A média de veículos nas {n_cidades} cidades é de:\n -> {media_v:.2f}" )
print( f"A média de acidentes em cidades com menos de 2000 veículos de passeio é de:\n -> {media_a:.2F}" )

Digite o código da cidade: SP
Digite o número de veículos de passeio (em 1999): 564
Digite o número de acidentes de trânsito com vítimas (em 1999): 544
Digite o código da cidade: AS
Digite o número de veículos de passeio (em 1999): 345
Digite o número de acidentes de trânsito com vítimas (em 1999): 154
Digite o código da cidade: VC
Digite o número de veículos de passeio (em 1999): 3254
Digite o número de acidentes de trânsito com vítimas (em 1999): 401
[('SP', 564, 544), ('AS', 345, 154), ('VC', 3254, 401)]
O maior índice de acidentes de trânsito é de:
 ->544 veículos em SP
O menor índice de acidentes de trânsito é de:
 ->154 veículos em AS
A média de veículos nas 3 cidades é de:
 ->1387.67
A média de acidentes em cidades com menos de 2000 veículos de passeio é de:
 ->349.00


24 - Faça um programa que receba o valor de uma dívida e mostre uma tabela com os seguintes dados: valor da dívida, valor dos juros, quantidade de parcelas e valor da parcela.
Os juros e a quantidade de parcelas seguem a tabela abaixo:
Quantidade de Parcelas  % de Juros sobre o valor inicial da dívida

1       0
3       10
6       15
9       20
12      25

Exemplo de saída do programa:

Valor da Dívida Valor dos Juros Quantidade de Parcelas  Valor da Parcela
R$ 1.000,00     0               1                       R$  1.000,00
R$ 1.100,00     100             3                       R$    366,00
R$ 1.150,00     150             6                       R$    191,67



In [87]:
divida = float( input( "Digite o valor da dívida: " ) )
valor=0.0
juros = 0.0
v_parcelo = 0.0
parcela_juros = [( 1,0 ), ( 3,10 ), ( 6,15 ), ( 9,20 ), ( 12,25 )]
print( "Valor da Dívida Valor dos Juros Quantidade de Parcelas  Valor da Parcela" )
for parcela,y in parcela_juros:
    juros = y/100*divida
    valor = divida + juros
    v_parcela = valor/parcela
    print( 
        f" R$ {valor:10.2f}   R$ {juros:10.2f}   {parcela:3}                     R$ {v_parcela:10.2f}"
     )

Digite o valor da dívida: 1000
Valor da Dívida Valor dos Juros Quantidade de Parcelas  Valor da Parcela
 R$    1000.00   R$       0.00     1                     R$    1000.00
 R$    1100.00   R$     100.00     3                     R$     366.67
 R$    1150.00   R$     150.00     6                     R$     191.67
 R$    1200.00   R$     200.00     9                     R$     133.33
 R$    1250.00   R$     250.00    12                     R$     104.17


25 - Faça um programa que mostre os n termos da Série a seguir:
S = 1/1 + 2/3 + 3/5 + 4/7 + 5/9 + ... + n/m. 
Imprima no final a soma da série.

In [111]:
z = int(input("Digite a quantidade de divisões que deseja somar: "))
sum = 0
n = [x for x in range(1,z+1)]
m = [x for x in range(1,(z*2)+1) if x%2!=0]
r = ''
lista = list(zip(n,m))
for n,m in lista:
    sum += n/m
    r += f"{n}/{m} + "
# print(n,m)
# print(lista)
print(f"S = {r[:len(r)-3]}")
print(f"A soma é = {sum}")

Digite a quantidade de divisões que deseja somar: 10
S = 1/1 + 2/3 + 3/5 + 4/7 + 5/9 + 6/11 + 7/13 + 8/15 + 9/17 + 10/19
A soma é = 6.066627765079777
